### DISCUSSION TITLE
Support clang option --gcc-toolchain/-gcc-toolchain

### DISCUSSION THREAD

**Discussion by MizuhaHimuraki** [ccache/ccache#837] [Development]
Hi, guys.

I have some problems with an embedded compiler project. Although I added the option '-fdebug-prefix-map', the preprocessor still uses the absolute path.

This option looks like 'isystem' or 'isysroot', here is the description copied from the Clang documentation

```
--gcc-toolchain=<arg>, -gcc-toolchain <arg>
Search for GCC installation in the specified directory on targets which commonly use GCC. The directory usually contains ‘lib{,32,64}/gcc{,-cross}/$triple’ and ‘include’. If specified, sysroot is skipped for GCC detection. Note: executables (e.g. ld) used by the compiler are not overridden by the selected GCC installation
```

Could I add the following code to the Compopt.cpp? Or is there a better solution?

> {"--gcc-toolchain=",        AFFECTS_CPP | TAKES_CONCAT_ARG | TAKES_PATH},
{"-gcc-toolchain",        AFFECTS_CPP | TAKES_ARG | TAKES_PATH},

**Comment by jrosdahl** [comment]
> I have some problems with an embedded compiler project. Although I added the option '-fdebug-prefix-map', the preprocessor still uses the absolute path.

I assume you mean that you have also set `base_dir` to make ccache rewrite absolute paths.

> Could I add the following code to the Compopt.cpp? Or is there a better solution?
>
> > {"--gcc-toolchain=",        AFFECTS_CPP | TAKES_CONCAT_ARG | TAKES_PATH},
> > {"-gcc-toolchain",        AFFECTS_CPP | TAKES_ARG | TAKES_PATH},

Sounds good, except that I don't think that `AFFECTS_CPP` should be set since (I guess) the option not only affects preprocessing.
